### PR TITLE
MB-48564: Update Sync Gateway Offers for 2.8.3

### PR DESCRIPTION
**Pull request by malscent**
*  Updated install script reference
*  Added 2.8.3 to available options in AWS
*  Added 2.8.3 to available options in Azure
*  Added 2.8.3 to available options in GCP
*  Updated AWS sync gateway ami for tagging control
*  Updated Tagging Documentation in couchbase-ami-creation for AWS

### CODE REVIEW

File: aws/CouchbaseSyncGateway/makeArchives.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        n) output_template_name=${OPTARG};;
+        i) input_template=${OPTARG};;
+        *) exit 1;;
+    esac
+done
+
+echo "In make archives"
+mkdir -p "${output}"
+
+cat "${input_template}" | jq --argjson map "$(cat "$mapping_file" | jq -r)" '.Mappings = $map' > "${output}${output_template_name}"
```
Comment (patrick-stephens) [inline aws/CouchbaseSyncGateway/makeArchives.sh:17]: Do we need to do anything to ensure the output is a valid location? Needs a / or something?

File: aws/couchbase-ami-creation/gateway-startup.sh [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+CONFIG=$(jq -r 'select(.Key == "couchbase:gateway:config").Value' <<< "$TAGS")
+BUCKET=$(jq -r 'select(.Key == "couchbase:gateway:bucket").Value' <<< "$TAGS")
+STACK_NAME=$(jq -r 'select(.Key == "aws:cloudformation:stack-name").Value' <<< "$TAGS")
+RESOURCE=$(jq -r 'select(.Key == "aws:cloudformation:logical-id").Value' <<< "$TAGS") 
+TAGGED_USERNAME=$(jq -r 'select(.Key == "couchbase:gateway:username").Value' <<< "$TAGS")
+TAGGED_PASSWORD=$(jq -r 'select(.Key == "couchbase:gateway:password").Value' <<< "$TAGS")
+NAME=$(jq -r 'select(.Key == "Name").Value' <<< "$TAGS")
+
+
+# if no version.. use built in version
```
Comment (patrick-stephens) [inline aws/couchbase-ami-creation/gateway-startup.sh:52]: too many full stops

File: aws/couchbase-ami-creation/gateway-startup.sh [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+# Connection string should be param if passed
+if [[ -n "$CONNECTION_PARAM" ]]; then
+    CONNECTION_STRING=$(aws ssm get-parameter --name "$CONNECTION_PARAM" --region "$region" | jq -r '.Parameter.Value')
+fi
+
+
+# if we don't have a connection string we need some default.. so we'll default to localhost
+if [[ -z "$CONNECTION_STRING" ]]; then
+    CONNECTION_STRING="couchbase://localhost"
```
Comment (patrick-stephens) [inline aws/couchbase-ami-creation/gateway-startup.sh:65]: do we need to support couchbases://?

File: aws/couchbase-ami-creation/gateway-startup.sh [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+      "num_index_replicas":0
+    }
+  }
+} 
+_EOF
+else
+    aws s3 cp "$CONFIG" "/opt/sync_gateway/etc/sync_gateway.json"
+fi
+
+
```
Comment (patrick-stephens) [inline aws/couchbase-ami-creation/gateway-startup.sh:146]: Remove line

File: aws/couchbase-ami-creation/gateway-startup.sh [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+      SUCCESS=$?
+   fi
+else
+   # Remove existing
+   rpm -e "$(rpm -qa | grep couchbase)"
+   rm -rf /opt/couchbase-sync-gateway/
+   # Update /etc/profile.d/couchbaseserver.sh
+    echo "#!/usr/bin/env sh
+export COUCHBASE_GATEWAY_VERSION=$VERSION" > /etc/profile.d/couchbaseserver.sh
+   bash /setup/couchbase_installer.sh -ch "http://localhost:8091" -u "$USERNAME" -p "$PASSWORD" -v "$VERSION" -os AMAZON -e AWS -c -d -g
```
Comment (patrick-stephens) [inline aws/couchbase-ami-creation/gateway-startup.sh:165]: Probably a general do we need to support the secure-only port soon?

File: aws/couchbase-ami-creation/gateway-startup.sh [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+ 
+fi
+
+
+
+if [[ -n "$RESOURCE" ]] && [[ -n "$STACK_NAME" ]]; then
+    echo "We need to notify $RESOURCE that $STACK_NAME is complete"
+    if [[ "$SUCCESS" == "0" ]]; then
+    # Calls back to AWS to signify that installation is complete
+    /opt/aws/bin/cfn-signal -e 0 --stack "$STACK_NAME" --resource "$RESOURCE" --region "$region"
```
Comment (patrick-stephens) [inline aws/couchbase-ami-creation/gateway-startup.sh:176]: inconsistent indentation

File: aws/CouchbaseSyncGateway/deploy.sh [hunk 7]
```diff
@@ -60,10 +57,10 @@ echo "Default: $SYNC_GATEWAY_INSTANCE_COUNT_DEFAULT"
 echo "GatewayVersion: $SYNC_GATEWAY_VERSION"
 echo "Default: $SYNC_GATEWAY_VERSION_DEFAULT"
 
-VPC_NAME=$(aws ec2 describe-vpcs --filter "Name=isDefault,Values=true" | jq -r '.Vpcs[].VpcId')
-#VpcName=vpc-0c1cd329084365f10
-SUBNET_ID=$(aws ec2 describe-subnets --filter "Name=vpc-id,Values=${VPC_NAME}" --max-items 2 --region "$REGION" | jq -r '.Subnets[].SubnetId' | paste -s -d ',' - | sed 's~,~\\,~g' )
-#SubnetId=subnet-08476a90d895839b4
+#VPC_NAME=$(aws ec2 describe-vpcs --filter "Name=isDefault,Values=true" --region "$REGION" | jq -r '.Vpcs[].VpcId')
```
[the remark this answers is not in the document]
Comment (malscent) [reply aws/CouchbaseSyncGateway/deploy.sh:60]: ahh damn

File: aws/couchbase-ami-creation/gateway-startup.sh (hunk 3 above)
Comment (malscent) [reply aws/couchbase-ami-creation/gateway-startup.sh:65]: eventually, but i have to figure out how I want to allow the user to specify the mount that contains the certs

File: aws/couchbase-ami-creation/gateway-startup.sh (hunk 5 above)
Comment (malscent) [reply aws/couchbase-ami-creation/gateway-startup.sh:165]: Yeah.. It's on my todo list

File: aws/couchbase-ami-creation/gateway-startup.sh [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+fi
+
+
+# if we don't have a connection string we need some default.. so we'll default to localhost
+if [[ -z "$CONNECTION_STRING" ]]; then
+    CONNECTION_STRING="couchbase://localhost"
+fi
+
+if [[ ! "$CONNECTION_STRING" =~ ^couchbase*|^http* ]]; then
+    CONNECTION_STRING="couchbase://${CONNECTION_STRING}"
```
Comment (patrick-stephens) [inline aws/couchbase-ami-creation/gateway-startup.sh:69]: It might be worth logging the old one but not a blocker

**Review by patrick-stephens — APPROVED** [review]

**Pull request merged**